### PR TITLE
Add sections about creating and modifying agent and proxy entities

### LINKED ISSUE #2597: Document breaking agent contract for 6.0 entity configuration
### Description
See https://github.com/sensu/sensu-go/blob/master/CHANGELOG.md#breaking:
> After initial creation, sensu-agent entity configuration cannot be changed
by modifying the agent's configuration file.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
I'm not sure what the entire scope is, but we'll probably need notes about agent config and websocket contracts.

### Additional context
- https://github.com/sensu/sensu-go/commit/c3914d6787dccd0c68a729bb9dc835c0952f557a
- [Proposal: Agent Entity Backend API Management](https://docs.google.com/document/d/1LhuWC-WUswrUigzKOqsNZrBT5sTdKhvpXlpcUFfF5DE/edit#heading=h.letafivq6wet)

cc @echlebek

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
In the entity reference, adds sections for:
- Create and modify agent entities
- Create and modify proxy entities

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2597

### CODE REVIEW

File: content/sensu-go/6.0/reference/entities.md [hunk 1]
```diff
@@ -66,6 +66,32 @@ You can modify the proxy entity later if needed.
 
 Use [proxy entity filters][19] to establish a many-to-many relationship between agent entities and proxy entities if you want even more power over the grouping.
 
+## Create and modify agent entities
+
+When an agent connects to a backend, the agent entity defnition is created from the information in `/etc/sensu/agent.yml`.
```
Comment (nikkictl) [inline content/sensu-go/6.0/reference/entities.md]: I would maybe say "is created from the agent.yml config file," with a link to where the default file location is on each OS. ex. linux: `/etc/sensu/agent`, freebsdb: `/usr/local/etc/sensu`, windows:`C:\ProgramData\sensu\config` or something like that?

File: content/sensu-go/6.0/reference/entities.md [hunk 2]
```diff
@@ -66,6 +66,32 @@ You can modify the proxy entity later if needed.
...
+
+You can modify agent entities with [sensuctl][32], the [entities API][31], and the [web UI][33].
+
+{{% notice note %}}
+**NOTE**: You cannot modify agent entities with `/etc/sensu/agent.yml`.
+The entity attributes in `/etc/sensu/agent.yml` are used only for initial entity creation.
```
Comment (nikkictl) [inline content/sensu-go/6.0/reference/entities.md]: Same thing here, `/etc/sensu/agent.yml` is only the _default_ config file path for linux.

File: content/sensu-go/6.0/reference/entities.md [hunk 3]
```diff
@@ -66,6 +66,32 @@ You can modify the proxy entity later if needed.
...
+{{% /notice %}}
+
+To maintain agent entities based on `/etc/sensu/agent.yml`, create ephemeral agent entities with the [deregister attribute][34] set to `true`.
+With this setting, the agent entity will deregister every time the agent process stops.
+When it restarts, it will revert to the original configuration from `/etc/sensu/agent.yml`
+You must set `deregister: true` in `/etc/sensu/agent.yml` before the agent entity is created.
```
Comment (nikkictl) [inline content/sensu-go/6.0/reference/entities.md]: Again, we should be more generic when referring to the agent config file.

File: content/sensu-go/6.0/reference/entities.md [hunk 4]
```diff
@@ -66,6 +66,32 @@ You can modify the proxy entity later if needed.
...
+
+If you delete an agent entity that you modified with sensuctl, the entities API, or the web UI, it will revert to the original configuration from `/etc/sensu/agent.yml`.
```
Comment (nikkictl) [inline content/sensu-go/6.0/reference/entities.md]: This line too!

File: content/sensu-go/6.0/reference/entities.md (hunk 1 above)
Comment (hillaryfraley) [reply content/sensu-go/6.0/reference/entities.md]: I'm only including the OS platforms we have supported packages for

Comment (hillaryfraley) [inline content/sensu-go/6.0/reference/entities.md]:
```suggestion
When an agent connects to a backend, the agent entity definition is created from the information in the `agent.yml` configuration file.
The default `agent.yml` file location depends on your operating system:

- Linux and macOS: `/etc/sensu/agent`
- Windows: `C:\ProgramData\sensu\config`
```

File: content/sensu-go/6.0/reference/entities.md [hunk 5]
```diff
@@ -66,6 +66,36 @@ You can modify the proxy entity later if needed.
 
 Use [proxy entity filters][19] to establish a many-to-many relationship between agent entities and proxy entities if you want even more power over the grouping.
 
+## Create and modify agent entities
+
+When an agent connects to a backend, the agent entity definition is created from the information in the `agent.yml` configuration file.
+The default `agent.yml` file location depends on your operating system:
+
+- Linux and macOS: `/etc/sensu/agent`
+- Windows: `C:\ProgramData\sensu\config`
```
Comment (nikkictl) [inline content/sensu-go/6.0/reference/entities.md]: Maybe we can point them to the `config-file` attribute in https://docs.sensu.io/sensu-go/latest/reference/agent/#general-configuration-flags instead. That has all the default file paths documented already!

File: content/sensu-go/6.0/reference/entities.md [hunk 6]
```diff
@@ -66,6 +66,36 @@ You can modify the proxy entity later if needed.
...
+
+You can modify agent entities with [sensuctl][32], the [entities API][31], and the [web UI][33].
+
+{{% notice note %}}
+**NOTE**: You cannot modify agent entities with the `agent.yml` configuration file.
```
Comment (nikkictl) [inline content/sensu-go/6.0/reference/entities.md]: ..unless you delete the agent entity

File: content/sensu-go/6.0/reference/entities.md [hunk 7]
```diff
@@ -66,6 +66,36 @@ You can modify the proxy entity later if needed.
...
 
+## Create and modify agent entities
+
+When an agent connects to a backend, the agent entity definition is created from the information in the `agent.yml` configuration file.
+The default `agent.yml` file location depends on your operating system:
+
+- Linux and macOS: `/etc/sensu/agent`
+- Windows: `C:\ProgramData\sensu\config`
+
+You can modify agent entities with [sensuctl][32], the [entities API][31], and the [web UI][33].
```
Comment (nikkictl) [inline content/sensu-go/6.0/reference/entities.md]:
```suggestion
You can manage agent entities via the backend with [sensuctl][32], the [entities API][31], and the [web UI][33].
```

File: content/sensu-go/6.0/reference/entities.md [hunk 8]
```diff
@@ -66,6 +66,36 @@ You can modify the proxy entity later if needed.
...
+
+{{% notice note %}}
+**NOTE**: You cannot modify agent entities with the `agent.yml` configuration file.
+The entity attributes in `agent.yml` are used only for initial entity creation.
```
Comment (nikkictl) [inline content/sensu-go/6.0/reference/entities.md]: ..unless you delete the agent entity

File: content/sensu-go/6.0/reference/entities.md [hunk 9]
```diff
@@ -66,6 +66,36 @@ You can modify the proxy entity later if needed.
...
+{{% /notice %}}
+
+To maintain agent entities based on `agent.yml`, create ephemeral agent entities with the [deregister attribute][34] set to `true`.
+With this setting, the agent entity will deregister every time the agent process stops.
```
Comment (nikkictl) [inline content/sensu-go/6.0/reference/entities.md]:
```suggestion
With this setting, the agent entity will deregister every time the agent process stops and its keepalive expires.
```

**Review by nikkictl — APPROVED** [review]
Thanks so much for applying the feedback, this looks really good! @echlebek anything else to add here?

**Pull request merged**